cwms-data-api/src/main/java/cwms/cda/data/dao/basin/BasinDao.java

```diff
@@ -28,41 +28,54 @@
 import cwms.cda.data.dao.JooqDao;
 import cwms.cda.data.dto.CwmsId;
 import cwms.cda.data.dto.basin.Basin;
-import cwms.cda.api.enums.UnitSystem;
-import cwms.cda.api.enums.Unit;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import org.jooq.Configuration;
+import org.jooq.DSLContext;
 import org.jooq.impl.DSL;
 import usace.cwms.db.dao.util.OracleTypeMap;
 import usace.cwms.db.jooq.codegen.packages.CWMS_BASIN_PACKAGE;
+import usace.cwms.db.jooq.codegen.packages.CWMS_UTIL_PACKAGE;
 import usace.cwms.db.jooq.codegen.packages.cwms_basin.RETRIEVE_BASIN;
 
-import java.sql.SQLException;
-import java.sql.ResultSet;
-import java.util.List;
-import java.util.ArrayList;
-
-import org.jooq.DSLContext;
-
 public class BasinDao extends JooqDao<Basin> {
 
     public BasinDao(DSLContext dsl) {
         super(dsl);
     }
 
-    public List<Basin> getAllBasins(String unitSystem, String officeId) {
+    public List<Basin> getAllBasins(String officeId, String unitSystem) {
         return connectionResult(dsl, c -> {
             setOffice(c, officeId);
-            String areaUnitIn = UnitSystem.EN.value().equals(unitSystem)
-                    ? Unit.SQUARE_MILES.getValue() : Unit.SQUARE_KILOMETERS.getValue();
-            try(ResultSet rs = CWMS_BASIN_PACKAGE.call_CAT_BASINS(DSL.using(c).configuration(), null, null, null, areaUnitIn, officeId).intoResultSet()) {
+            // possibly call another procedure to get the units
+            Configuration configuration = getDslContext(c, officeId).configuration();
+            String areaUnitIn = unitSystem.compareToIgnoreCase("SI") == 0
```

Comment:
Use the unit system enum values for comparison given they are constants

Reply:
Replaced with enums